sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -1392,6 +1390,117 @@ def test_enet_sample_weight_sparse():
         reg.fit(X, y, sample_weight=sw, check_input=True)
 
 
+@pytest.mark.parametrize('use_weights_in_cv', [True, False])
+@pytest.mark.parametrize('fit_intercept', [True, False])
+def test_enet_cv_sample_weight_correctness(use_weights_in_cv, fit_intercept):
+    """Test that ElasticNetCV with sample weights gives correct results."""
+    rng = np.random.RandomState(42)
+    n_splits, n_samples, n_features = 3, 10, 5
+    X = rng.rand(n_splits * n_samples, n_features)
+    beta = rng.rand(n_features)
+    beta[0:2] = 0
+    y = X @ beta + rng.rand(n_splits * n_samples)
+    sw = np.ones_like(y)
+
+    # Set alphas, otherwise the two cv models might use different ones.
+    if fit_intercept:
+        alphas = np.linspace(0.001, 0.01, num=91)
+    else:
+        alphas = np.linspace(0.01, 0.1, num=91)
+
+    # We weight the first fold 2 times more.
+    sw[:n_samples] = 2
+    groups_sw = np.r_[np.full(n_samples, 0), np.full(n_samples, 1),
+                      np.full(n_samples, 2)]
+    splits_sw = list(LeaveOneGroupOut().split(X, groups=groups_sw))
+    reg_sw = ElasticNetCV(alphas=alphas, cv=splits_sw,
+                          fit_intercept=fit_intercept,
+                          use_weights_in_cv=use_weights_in_cv)
+    reg_sw.fit(X, y, sample_weight=sw)
+
+    # We repeat the first fold 2 times and provide splits ourselves
+    X = np.r_[X[:n_samples], X]
+    y = np.r_[y[:n_samples], y]
+    groups = np.r_[np.full(2*n_samples, 0), np.full(n_samples, 1),
+                   np.full(n_samples, 2)]
+    splits = list(LeaveOneGroupOut().split(X, groups=groups))
+    reg = ElasticNetCV(alphas=alphas, cv=splits, fit_intercept=fit_intercept,
+                       use_weights_in_cv=use_weights_in_cv)
+    reg.fit(X, y)
+
+    # ensure that we chose meaningful alphas, i.e. not boundaries
+    assert alphas[0] < reg.alpha_ < alphas[-1]
+    assert reg_sw.alpha_ == reg.alpha_
+    assert_allclose(reg_sw.coef_, reg.coef_)
+    if fit_intercept is not None:
+        assert reg_sw.intercept_ == pytest.approx(reg.intercept_)
+    if use_weights_in_cv:
+        # Note: The order of groups from LeaveOneGroupOut is the same for both.
+        assert_allclose(reg_sw.mse_path_, reg.mse_path_)
+
+
+# TODO: Compare ElasticNetCV to GridSearchCV
```

Comment:
How would you compare them exactly? 🙂
Would you compare the results of `ElasticNetCV` vs the ones of `GridSearchCV(ElasticNet)` on the new behavior introduced?

Reply:
I added a test for that in https://github.com/scikit-learn/scikit-learn/pull/16449/commits/49a143b7c62b87c9b78638a02e256d3dd49af7e7.